### PR TITLE
Added security options

### PR DESCRIPTION
**Pull request by ltangvald**
* Added option for randomly generated root password
* Added option to immediately expire root password
---
Installs pwgen package
MYSQL_RANDOM_ROOT_PASSWORD:
Uses pwgen to generate a password for the server's root user, and outputs it to the containers stdout

MYSQL_ONETIME_PASSWORD:
Sets root user as expired once init is complete, forcing a password change on first login (5.6+)

### CODE REVIEW

File: 5.6/docker-entrypoint.sh [hunk 1]
```diff
@@ -11,12 +11,11 @@ if [ "$1" = 'mysqld' ]; then
...
-			echo >&2 'error: database is uninitialized and MYSQL_ROOT_PASSWORD not set'
-			echo >&2 '  Did you forget to add -e MYSQL_ROOT_PASSWORD=... ?'
-			exit 1
-		fi
-
+		if [ -z "$MYSQL_ROOT_PASSWORD" -a -z "$MYSQL_ALLOW_EMPTY_PASSWORD" -a -z "$MYSQL_RANDOM_ROOT_PASSWORD" ]; then
+                        echo >&2 'error: database is uninitialized and password option is not specified '
+                        echo >&2 '  You need to specify one of MYSQL_ROOT_PASSWORD, MYSQL_ALLOW_EMPTY_PASSWORD and MYSQL_RANDOM_ROOT_PASSWORD'
+                        exit 1
+                fi
```
Comment (tianon) [inline 5.6/docker-entrypoint.sh]: Whitespace here seems a little off -- is this tabs, or spaces?

**Comment by tianon** [comment]
Very nice additions IMO :+1: :metal:

File: 5.6/docker-entrypoint.sh (hunk 1 above)
Comment (ltangvald) [inline 5.6/docker-entrypoint.sh]: Yeah, looks like a mix. Will fix :)

**Comment by ltangvald** [comment]
Should be fixed now :)

**Comment by yosifkit** [comment]
LGTM

**Comment by tianon** [comment]
LGTM

**Pull request merged**